src/daemon/XorgDisplayServer.cpp

```diff
@@ -88,28 +89,55 @@ namespace SDDM {
     }
 
     void XorgDisplayServer::addCookie(const QString &file) {
-        // log message
+        Xauth auth = { 0 };
+        char localhost[HOST_NAME_MAX + 1] = { 0 };
+
         qDebug() << "Adding cookie to" << file;
 
-        // Touch file
-        QFile file_handler(file);
-        file_handler.open(QIODevice::Append);
-        file_handler.close();
+        if (gethostname(localhost, HOST_NAME_MAX) < 0) {
+            strcpy(localhost, "localhost");
+        }
 
-        QString cmd = QStringLiteral("%1 -f %2 -q").arg(mainConfig.X11.XauthPath.get()).arg(file);
+        // libXau expects binary data, not a string
+        QByteArray cookieBinary = QByteArray::fromHex(m_cookie.toLatin1());
+
+        // set up the auth entry
+        char cookieName[] = "MIT-MAGIC-COOKIE-1";
+        auth.family = FamilyLocal;
+        auth.address = localhost;
+        auth.address_length = strlen(auth.address);
+        auth.name = cookieName;
+        auth.name_length = sizeof(cookieName);
+        auth.data_length = cookieBinary.count();
+        auth.data = cookieBinary.data();
+
```

Comment:
Missing initialization of `auth.number{,_length}`, intentional? It should be filled with `m_display` (excluding the `:`).